spec/src/main/asciidoc/configexamples.asciidoc

```diff
@@ -129,3 +129,153 @@ public class InjectedConfigUsageSample {
+If any of the property is missing and there is neither default value nor property is not optional, DeploymentException will be thrown. 
+In order to avoid this, you can supply a default value when defining the field. Alternatively, you can use `@ConfigProperty` to provide a default value. 
+You can also use `Optional<T>` or OptionalInt, OptionalDouble, OptionalLong as the type.
+If any of the property value cannot be converted to the specified type, DeploymentException will be thrown as well.
+
+* programmatic look up via `Config.getConfigProperties()`
+[source, java]
+----
+Config config = ConfigProvider.getConfig();
+Details serverDetails = config.getConfigProperties(Details.class);
+Details clientDetails = config.getConfigProperties(Details.class, "client"); 
+Details details = config.getConfigProperties(Details.class, ""); //use no prefix
+----
+
+In the above example, the `serverDetails` will contain the following info, because the prefix `server` on the bean class `Details` is in action:
+
+```
+serverDetails.host -> server.host -> localhost
+serverDetails.port -> server.port -> 9080
+serverDetails.endpoint -> server.endpoint -> query
+serverDetails.getLocation() -> server.old.location -> London
+```
+The `clientDetails` will retrieve the following properties, because the specified prefix `client` overrides the prefix on the bean class.
+
+```
+clientDetails.host -> client.host -> myHost
+clientDetails.port -> client.port -> 9081
+clientDetails.endpoint -> client.endpoint -> shelf
+clientDetails.getLocation() -> client.old.location -> Dublin
+```
+The `empty` prefix overrides the prefix set on the bean class `server`. Therefore, this will retrieve the following properties.
+
+```
+details.host -> host -> anotherHost
+details.port -> port -> 9082
+details.endpoint -> endpoint -> book
+details.getLocation() -> old.location -> Berlin
+```
+In the above programmatic lookup, the configuration properties class must contain a zero-arg constructor. Otherwise, `java.lang.IllegalArgumentException` will be thrown. 
+
+If any of the property cannot be found and there is neither default value nor property is not optional, `java.util.NoSuchElementException` will be thrown. 
+In order to avoid this, you can supply a default value when defining the field. Alternatively, you can use `@ConfigProperty` to provide a default value. 
+You can also use `Optional<T>` or OptionalInt, OptionalDouble, OptionalLong as the type. If any property values cannot be converted to the specified type, `java.lang.IllegalArgumentException` will be thrown.
+
+When using programmatic lookup of the configuration properties, the configuration properties class does not need to be annotated with 
+`@ConfigProperties`, though it is perfectly ok to contain that annotation as shown above. However, when using injection of the configuration properties, 
```

Comment:
Clarify that the annotation can be used programatically to set the prefix?

Reply:
done